solidity-v1/dashboard/src/pages/applications/ThresholdApplicationPage.jsx

```diff
@@ -0,0 +1,128 @@
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.AuthorizeAndStakeOnThreshold, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: false,
+      })
+    },
+    [openModal]
+  )
+
+  const stakeToT = useCallback(
+    async (data, awaitingPromise) => {
+      const {
+        operatorAddress,
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.StakeOnThresholdWithoutAuthorization, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: true,
+      })
+    },
+    [openModal]
+  )
+
+  const thresholdAuthData = useMemo(() => {
+    const thresholdData = thresholdAuthState.authData.filter((dataObj) => {
+      return !dataObj.isStakedToT || !dataObj.contracts[0].isAuthorized
+    })
+    if (!selectedOperator.operatorAddress) {
+      return thresholdData
+    }
+    return thresholdData.filter((data) =>
+      isSameEthAddress(data.operatorAddress, selectedOperator.operatorAddress)
+    )
+  }, [selectedOperator.operatorAddress, thresholdAuthState.authData])
+
+  const authorizationHistoryData = useMemo(() => {
+    if (!selectedOperator.operatorAddress)
+      return thresholdAuthState.authData
+        .filter((authData) => authData.contracts[0].isAuthorized)
+        .map(toAuthHistoryData)
+    return thresholdAuthState.authData
+      .filter(
+        ({ operatorAddress, contracts }) =>
+          contracts[0].isAuthorized &&
+          isSameEthAddress(operatorAddress, selectedOperator.operatorAddress)
+      )
+      .map(toAuthHistoryData)
+  }, [thresholdAuthState.authData, selectedOperator.operatorAddress])
+
+  return (
+    <>
+      <LoadingOverlay
+        isFetching={thresholdAuthState.isFetching}
+        skeletonComponent={
+          <DataTableSkeleton columns={4} subtitleWidth="40%" />
+        }
+      >
+        <AuthorizeStakesBanner
+          numberOfStakesToAuthorize={thresholdAuthData.length}
+        />
+        <AuthorizeThresholdContracts
+          filterDropdownOptions={thresholdAuthState.authData}
+          onSelectOperator={setOperator}
+          selectedOperator={selectedOperator}
+          data={thresholdAuthData}
+          onAuthorizeBtn={authorizeContract}
+          onStakeBtn={stakeToT}
+        />
+      </LoadingOverlay>
+      <LoadingOverlay
+        isFetching={thresholdAuthState.isFetching}
+        skeletonComponent={<DataTableSkeleton columns={4} subtitleWidth="0" />}
+      >
+        <ThresholdAuthorizationHistory contracts={authorizationHistoryData} />
+      </LoadingOverlay>
+    </>
+  )
+}
+
+const toAuthHistoryData = (authData) => ({
+  ...authData,
+  ...authData.contracts[0],
```

Comment:
I think we can convert `contracts` field to `contract` object in redux state data because there is only one contract to authorize. eg:
```
contract: {
 contractName,
 operatorContractAddress
}
```